src/qililab/analog/annealing_program.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Any, Callable
+
+import numpy as np
+
+from qililab.platform.components import Bus
+from qililab.waveforms import Arbitrary as ArbitraryWave
+
+
+@dataclass
+class AnnealingProgram:
+    """Class for an Annealing Program. The program should have the format
+
+    [
+        {"qubit_0": {"sigma_x" : 0, "sigma_y" : 1, "sigma_z" : 2},
+         "coupler_1_0 : {...},
+        },      # time=0ns
+        {...},  # time=1ns
+    .
+    .
+    .
+    ]
+    """
+
+    platform: Any
+    anneal_program: list[dict[str, dict[str, float]]]
```

Comment:
I chose to use a dictionary because i'm assuming that ultimately the annealing schedule is something that's gonna come from an upper layer outside qililab (either from the algorithms team library or from a notebook, etc. ), so I wanted to go for some easy to convert format that's serializable / can be sent through the net, easy to store, etc. So for that the dictionary type seemed ideal. Once it's more clear what the entry point (how we convert from a qibo hamiltonian to this) we can implement some more rigid class if relevant

Reply:
I understand what you are saying, it's probably unlikely that someone will create a schedule manually, but still each layer should be independent.

Also, having a concrete class with fields will help the algorithms team as well, since they will have a reference. (Imagine renaming "sigma_x" or altering something, right now we would have to manually relay the change to everyone)

Anyway, this is not a hard requirement, but I think the change will help the whole implementation of the flow.

Reply:
I've created an issue for this at https://linear.app/qilimanjaro/issue/QHC-689/create-entry-point-class-for-annealingprogram-instead-of-dictionary